pyro/distributions/hmm.py

```diff
@@ -506,14 +506,22 @@ def rsample(self, sample_shape=torch.Size()):
         obs_dim = self.obs_dim
         hidden_dim = self.hidden_dim
         trans = self._trans + self._obs.marginalize(right=self.obs_dim).event_pad(left=self.hidden_dim)
-        if trans.batch_shape[-1] != self.duration:
```

Comment:
Is the issue here that `batch_shape` might be empty?

Reply:
That's one issue. Another issue is that `trans` is a `Gaussian` and has multiple underlying tensors some of which may be expanded and others unexpanded. Some of the ops after this line require all those underlying tensors to be expanded to the same shape. We could fix those underlying ops, but that seemed difficult so instead I've added manual `x = x.expand(x.batch_shape)` in a few places.

Reply:
Thanks for explaining.